### PR TITLE
New tip shape :class:`.StealthTip`, allow specifying tip shape of :class:`.NumberLine`

### PR DESCRIPTION
**Pull request by behackl**
This adds a new and somewhat more fancy arrow tip shape, `StealthTip`, inspired from the [corresponding TikZ arrow tip](https://tikz.dev/tikz-arrows#sec-16.3).

This PR also allows properly propagating the tip shape to `Axes` and `Numberline`.

image

## Reviewer Checklist
- [ ] The PR title is descriptive enough for the changelog, and the PR is labeled correctly
- [ ] If applicable: newly added non-private functions and classes have a docstring including a short summary and a PARAMETERS section
- [ ] If applicable: newly added functions and classes are tested

### CODE REVIEW

**Comment by behackl** [comment]
@MrDiver what is the proper way to typehint with Type?

```py
from typing import Type

tip_shape: Type[ArrowTip] | None = None
```

or something along these lines?

**Comment by MrDiver** [comment]
> @MrDiver what is the proper way to typehint with Type?
> 
> ```python
> from typing import Type
> 
> tip_shape: Type[ArrowTip] | None = None
> ```
> 
> or something along these lines?

This is fine like it is, theres not really a different way to type `Type`

File: tests/test_graphical_units/test_axes.py [hunk 1]
```diff
@@ -14,6 +14,7 @@ def test_axes(scene):
         x_length=6,
         y_length=6,
         color=WHITE,
+        y_axis_config={"tip_shape": StealthTip},
```
Comment (MrDiver) [inline tests/test_graphical_units/test_axes.py:17]: This seems like a change in defaults if i am not mistaken?

**Review by MrDiver — CHANGES_REQUESTED** [review]

File: tests/test_graphical_units/test_axes.py (hunk 1 above)
Comment (behackl) [reply tests/test_graphical_units/test_axes.py:17]: I just changed the test case to have an x-axis with the default triangle tip, and the y-axis with the new stealth tip, it shouldn't change any existing default. (Plus, I committed the control data that has been changed as a result of it.)

What do you mean with change in defaults?

**Review by MrDiver — CHANGES_REQUESTED** [review]

File: manim/mobject/graphing/number_line.py [hunk 2]
```diff
@@ -140,6 +148,7 @@ def __init__(
         include_tip: bool = False,
         tip_width: float = DEFAULT_ARROW_TIP_LENGTH,
         tip_height: float = DEFAULT_ARROW_TIP_LENGTH,
+        tip_shape: type[ArrowTip] | None = None,
```
[the remark this answers is not in the document]
Comment (MrDiver) [reply manim/mobject/graphing/number_line.py:151]: If it doesn't cause an error in the other versions of python then we can just remove the unused import i guess.

[the remark this answers is not in the document]
Comment (behackl) [reply manim/mobject/graphing/number_line.py:151]: The current version doesn't have a `Type` input anymore, already done 👍

**Pull request merged**